### DISCUSSION TITLE
Allow to use ReactNode as labels for Select and MultiSelect

### DISCUSSION THREAD

**Discussion by finom** [mantinedev/mantine#7037] [Feature requests]
Thanks for the great work. It would be nice to have an option to pass `ReactNode` as a `label`. This would simplify development of simple custom features and would allow to avoid creating a custom component with `Combobox`.

```tsx
<MultiSelect
            data={[{ label: <>Hello</>, value: 'hello' }]}
...
```

**Comment by rtivital** [comment]
It is not possible, label is used for search, if you want to render a custom react node in the dropdown, use `renderOption`. If you would like to have more advanced customizations, use Combobox component.

**Reply by finom** [reply]
Indeed if `label` is used for search it's not possible to use `ReactNode`. Let me adjust my proposal:

1. Allow labels to be `ReactNode`
2. Add `searchTerm` property to select option that's going to be used for search if presented:
  - If `searchTerm` - use it for search
  - Else if `label` is a string  - use it for search
  - Else if `label` is a `ReactNode`  - never include the item in search results

It's not only going to save a lot of human-hours, but also preserves backward compatibility. Thanks.
